tests/integration/any_charm.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""This code snippet is used to be loaded into any-charm which is used for integration tests."""
+import os
+import pathlib
+import signal
+import subprocess
+import tempfile
+
+from any_charm_base import AnyCharmBase
+from ingress import IngressRequires
+
+
+class AnyCharm(AnyCharmBase):
+    """Execute a simple web-server charm to test the ingress-proxy relation"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.ingress = IngressRequires(
+            self,
+            {"service-hostname": self.app.name, "service-name": self.app.name, "service-port": 80},
+        )
+
+    def update_ingress(self, ingress_config):
+        self.ingress.update_config(ingress_config)
+
+    @staticmethod
+    def start_server(port: int = 80):
+        """Start an HTTP server daemon."""
+        www_dir = tempfile.mkdtemp()
+        # We create a pid file to avoid concurrent executions of the http server
+        pid_file = pathlib.Path("/tmp/any.pid")
+        if pid_file.exists():
+            os.kill(int(pid_file.read_text()), signal.SIGKILL)
+            pid_file.unlink()
+        p = subprocess.Popen(
```

Comment:
Do you think it is worth expanding on the variable name a bit? Maybe `http_proc` or something like that? `p` can mean lots of things and it requires the reader to refer back to the original definition to figure out what it actually is with extremely short variable names